index.d.ts

```diff
@@ -22,27 +23,31 @@ declare module 'react-emoji-render' {
     options?: Options;
   }
 
+  type RequireProperty<T, Prop extends keyof T> = T & {[key in Prop]-?:T[key]};
+
+  type PropsRequireTextOrChildren = RequireProperty<Props, 'text'> | RequireProperty<Props, 'children'>;
```

Comment:
Pretty clean 👌🏻 

The types were updated at https://github.com/tommoor/react-emoji-render/pull/99, so you will need to take into account these changes, but this approach is perfect.